libs/pika/topology/src/topology.cpp

```diff
@@ -1169,7 +1169,28 @@ namespace pika::threads::detail {
                 fmt::streamed(mask));
         }
 
-        main_thread_affinity_mask_ = std::move(mask);
+        // The mask is assumed to use physical/OS indices (as returned by e.g. hwloc-bind --get
+        // --taskset or taskset --pid) while pika deals with logical indices from this point
+        // onwards. We convert the mask from physical indices to logical indices before storing it.
+        mask_type logical_mask = mask_type();
```

Comment:
```suggestion
        auto logical_mask = mask_type();
```
?

Reply:
Updated to simply:
```suggestion
        mask_type logical_mask();
```

Reply:
Well, it turned into
```suggestion
        mask_type logical_mask{};
```
in the end because of the most vexing parse. But updated in any case.